gtsam/slam/dataset.cpp

```diff
@@ -541,10 +541,18 @@ std::map<Key, Pose3> parse3DPoses(const string& filename) {
 }
 
 /* ************************************************************************* */
-BetweenFactorPose3s parse3DFactors(const string& filename) {
+BetweenFactorPose3s parse3DFactors(const string& filename, 
+    const noiseModel::Diagonal::shared_ptr& corruptingNoise) {
   ifstream is(filename.c_str());
   if (!is) throw invalid_argument("parse3DFactors: can not find file " + filename);
 
+  // If asked, create a sampler with random number generator
+  Sampler sampler;
+  if (corruptingNoise) {
+    sampler = Sampler(corruptingNoise);
+  }
```

Comment:
@binitshah This is first constructing a sampler object, and then copying a new sampler object `Sampler(corruptingNoise)` to it. Build fails because there is no default constructor defined for Sampler, and so we cannot do `Sampler sampler;`. Could you consider removing these lines and instantiating the Sampler object before line 598 where it is being used?